### PR TITLE
PLAT-10533: Added User-Agent header in requests

### PR DESCRIPTION
**Pull request by symphony-elias**
### Ticket
PLAT-10533

### Description
Added user-agent header with BDK version and python version.

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [N/A] Docstrings added or updated
- [N/A] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

**Comment by symphony-youri** [comment]
Please provide a description in the commit with at least an example of the header (saying why we set it would be nice too)

File: symphony/bdk/core/client/api_client_factory.py [hunk 1]
```diff
@@ -71,11 +74,34 @@ def _get_api_client(self, server_config, context) -> ApiClient:
...
         if proxy_config.are_credentials_defined():
-            configuration.proxy_headers = urllib3.util.make_headers(proxy_basic_auth=proxy_config.get_credentials())
+            configuration.proxy_headers = urllib3.util.make_headers(proxy_basic_auth=proxy_config.get_credentials(),
+                                                                    user_agent=ApiClientFactory._user_agent())
+        else:
+            configuration.proxy_headers = urllib3.util.make_headers(user_agent=ApiClientFactory._user_agent())
+
+    @staticmethod
+    def _user_agent():
+        return f"Symphony-BDK-Python/{ApiClientFactory._bdk_version()} Python/{ApiClientFactory._python_version()}"
```
Comment (symphony-youri) [inline symphony/bdk/core/client/api_client_factory.py:122]: The format looks a bit different from the Java one:

return "Symphony BDK/" + getBdkVersion() + "/java/" + System.getProperty("java.version");

Comment (symphony-elias) [reply symphony/bdk/core/client/api_client_factory.py:122]: Yes but I realized we should not have spaces : https://tools.ietf.org/html/rfc7231#section-5.5.3
I was thinking this was more in line with the RFD. WDYT ? 
Either we keep like the java one or we change the user-agent in the java BDK

Comment (symphony-youri) [reply symphony/bdk/core/client/api_client_factory.py:122]: I agree, let's update it for the Java BDK then (and consider it a non breaking change...)

Comment (symphony-elias) [reply symphony/bdk/core/client/api_client_factory.py:122]: GH issue opened here on the Java BDK: https://github.com/finos/symphony-bdk-java/issues/471

**Review by symphony-youri — APPROVED** [review]

**Pull request merged**